server/datastore/mysql/activities.go

```diff
@@ -886,12 +934,65 @@ func (ds *Datastore) CancelHostUpcomingActivity(ctx context.Context, hostID uint
 	})
 
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// TODO: create canceled activity, this must be done via svc.NewActivity (not
-	// ds.NewActivity), so once canceled activities are implemented return the
-	// ready-to-insert activity struct to the caller and let svc do the rest.
+	// creating the canceled activity must be done via svc.NewActivity (not
+	// ds.NewActivity), so we return the ready-to-insert activity struct to the
+	// caller and let svc do the rest.
+	return pastAct, nil
+}
+
+func clearLockWipeForCanceledActivity(ctx context.Context, tx sqlx.ExtContext, hostID uint, executionID string) error {
+	const clearLockStmt = `DELETE FROM host_mdm_actions WHERE host_id = ? AND lock_ref = ?`
+	resLock, err := tx.ExecContext(ctx, clearLockStmt, hostID, executionID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "delete host_mdm_actions for lock")
+	}
+
+	const clearWipeStmt = `DELETE FROM host_mdm_actions WHERE host_id = ? AND wipe_ref = ?`
+	resWipe, err := tx.ExecContext(ctx, clearWipeStmt, hostID, executionID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "delete host_mdm_actions for wipe")
+	}
+
+	lockCnt, _ := resLock.RowsAffected()
+	wipeCnt, _ := resWipe.RowsAffected()
+	if lockCnt > 0 || wipeCnt > 0 {
+		// if it did deleted host_mdm_actions, then it was a lock or wipe activity,
+		// we need to deleted the "past" activity that gets created immediately
+		// when that command is queued.
```

Comment:
Nit:

```suggestion
		// if it did delete host_mdm_actions, then it was a lock or wipe activity,
		// we need to delete the "past" activity that gets created immediately
		// when that command is queued.
```

Reply:
Thanks! I'll fix it in the next small PR.